### PR TITLE
Settings manager: Create a class to manage all shared settings

### LINKED ISSUE #42: Create a singleton cpp class to handle settings

### LINKED ISSUE #205: Disable values inputs check in debug mode.

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #205 
Will help #199 
Fix #42

### CODE REVIEW

File: src/logger.cpp [hunk 1]
```diff
@@ -42,7 +34,7 @@ void Logger::installHandler()
     originalHandler = qInstallMessageHandler(handleMessage); // This function returns the previous message handler
 
     if (qEnvironmentVariableIsEmpty("QT_MESSAGE_PATTERN")) {
-        qSetMessagePattern(QStringLiteral("%{time [hh:mm:ss.zzz]} %{message}"));
```
Comment (jaxxzer) [inline src/logger.cpp]: Was adding this hunk intentional? What does this do?

File: src/logger.h [hunk 2]
```diff
@@ -115,7 +114,6 @@ class Logger : public QObject
     QFile _file;
     QTextStream _fileStream;
     QStringList _registeredCategories;
-    QSettings _settings;
```
Comment (jaxxzer) [inline src/logger.h]: Can we assign SettingsManager::self()->settings() to a member _settings, and use that everywhere for better readability?

File: src/logger.cpp (hunk 1 above)
Comment (patrickelectric) [reply src/logger.cpp]: Ops, my bad, I was using a personal pattern while debugging.

File: src/logger.h (hunk 2 above)
Comment (patrickelectric) [reply src/logger.h]: Yep, Will do that

**Pull request merged**